frontend/src/metabase/querying/components/FieldPanel/FieldPanel.unit.spec.tsx

```diff
@@ -112,26 +117,242 @@ describe("QueryColumnPicker", () => {
+    const [group, firstColumn, secondColumn] = screen.getAllByRole("checkbox");
+    expect(group).toBeChecked();
+    expect(group).toBeEnabled();
+
+    // remove all columns - the first column is not removed
+    await userEvent.click(group);
+    expect(group).not.toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeDisabled();
+    expect(secondColumn).not.toBeChecked();
+    expect(secondColumn).toBeEnabled();
+
+    // add all columns
+    await userEvent.click(group);
+    expect(group).toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeEnabled();
+    expect(secondColumn).toBeChecked();
+    expect(secondColumn).toBeEnabled();
+
+    // remove the first column - the second column is disabled
+    await userEvent.click(firstColumn);
+    expect(group).not.toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).not.toBeChecked();
+    expect(firstColumn).toBeEnabled();
+    expect(secondColumn).toBeChecked();
+    expect(secondColumn).toBeDisabled();
+
+    // add the first column
+    await userEvent.click(firstColumn);
+    expect(group).toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeEnabled();
+    expect(secondColumn).toBeChecked();
+    expect(secondColumn).toBeEnabled();
+
+    // remove the second column - the first column is disabled
+    await userEvent.click(secondColumn);
+    expect(group).not.toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeDisabled();
+    expect(secondColumn).not.toBeChecked();
+    expect(secondColumn).toBeEnabled();
+
+    // add the second column
+    await userEvent.click(secondColumn);
+    expect(group).toBeChecked();
+    expect(group).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeEnabled();
+    expect(secondColumn).toBeChecked();
+    expect(secondColumn).toBeEnabled();
+  });
+
+  it("should not allow to remove custom columns", async () => {
     const query = Lib.expression(
       createQuery(),
       -1,
       "Custom",
       Lib.expressionClause("+", [1, 2]),
     );
     setup({ query });
-    const [orderGroup] = screen.getAllByRole("checkbox");
+    const [orderGroup, firstColumn] = screen.getAllByRole("checkbox");
     const customColumn = screen.getByRole("checkbox", { name: "Custom" });
     expect(orderGroup).toBeChecked();
-    expect(orderGroup).toBeDisabled();
+    expect(orderGroup).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeEnabled();
     expect(customColumn).toBeChecked();
     expect(customColumn).toBeDisabled();
+
+    await userEvent.click(orderGroup);
+    expect(orderGroup).not.toBeChecked();
+    expect(orderGroup).toBeEnabled();
+    expect(firstColumn).toBeChecked();
+    expect(firstColumn).toBeDisabled();
```

Comment:
> Note that when there are expressions we now disallow deselecting the first field which is not the case in master. This is to prevent from construction of subsequently broken queries in the notebook editor which requires 1 field to be selected.

This comment deserves to live somewhere in code :thinking:

Reply:
Fixed